### PR TITLE
[17.09] Prevent reloading when data table is being modified

### PR DESCRIPTION
**Pull request by mvdbeek**
Fixes https://github.com/galaxyproject/galaxy/issues/4783.

### CODE REVIEW

File: lib/galaxy/tools/data_manager/manager.py [hunk 1]
```diff
@@ -317,10 +317,10 @@ def process_result(self, out_data):
...
-                                  'reload_tool_data_tables',
-                                  noop_self=True,
-                                  kwargs={'table_name': data_table_name})
+        send_control_task(self.data_managers.app,
+                          'reload_tool_data_tables',
+                          noop_self=True,
+                          kwargs={'table_name': data_table_name})
         if self.undeclared_tables and data_tables_dict:
             # We handle the data move, by just moving all the data out of the extra files path
             # moving a directory and the target already exists, we move the contents instead
```
Comment (nsoranzo) [inline lib/galaxy/tools/data_manager/manager.py]: Need to fix the same problem below at lines 341-343.

File: lib/galaxy/tools/data_manager/manager.py [hunk 2]
```diff
@@ -317,10 +317,10 @@ def process_result(self, out_data):
...
-                                  'reload_tool_data_tables',
-                                  noop_self=True,
-                                  kwargs={'table_name': data_table_name})
+            send_control_task(self.data_managers.app,
+                              'reload_tool_data_tables',
+                              noop_self=True,
+                              kwargs={'table_name': data_table_name})
         if self.undeclared_tables and data_tables_dict:
             # We handle the data move, by just moving all the data out of the extra files path
             # moving a directory and the target already exists, we move the contents instead
```
Comment (nsoranzo) [inline lib/galaxy/tools/data_manager/manager.py]: Need to fix the same problem below at lines 341-343.

**Review by nsoranzo — APPROVED** [review]

**Comment by dannon** [comment]
Looks good to me now!

**Comment by martenson** [comment]
@mvdbeek nice fix! :D

**Pull request merged**